src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,130 @@
+                }
+                return result;
+            case "mark": //command is mark 2
+                Task currTask = currList.getTaskAt(Integer.parseInt(parts[1]) - 1); //the arraylist is 0 indexed, so task 1 is actually 0 index
+                currTask.setDone();
+                result += "Let's go! I've marked this task as done:\n";
+                result += currTask;
+                return result;
+            case "unmark":
+                Task unmarkTask = currList.getTaskAt(Integer.parseInt(parts[1]) - 1); //the arraylist is 0 indexed, so task 1 is actually 0 index
+                unmarkTask.setUndone();
+                result += "Oh man! I've marked this task as undone:\n";
+                result += unmarkTask;
+                return result;
+            case "deadline": //can abstract this whole case to be generalized
+                String[] temp = input.split(" /by ", 2);
+                String by = temp[1];
+                String deadlineDesc = temp[0].split("deadline ")[1];
+                Deadline deadlineTask = new Deadline(deadlineDesc, LocalDate.parse(by));
+                currList.addTask(deadlineTask);
+                result += deadlineTask.addTaskMessage();
+                result += Ui.taskNumberMessage(currList);
+                return result;
+            case "event": //can abstract this whole case to be generalized
+                String[] temp1 = input.split(" /at ", 2);
+                String at = temp1[1];
+                String eventDesc = temp1[0].split("event ", 2)[1];
+                Event eventTask = new Event(eventDesc, LocalDate.parse(at));
+                currList.addTask(eventTask);
+                result += eventTask.addTaskMessage();
+                result += Ui.taskNumberMessage(currList);
+                return result;
+            case "todo": //can abstract this whole case to be generalized
+                String todoDesc = input.split("todo ")[1];
+                Todo todoTask = new Todo(todoDesc);
+                currList.addTask(todoTask);
+                result += todoTask.addTaskMessage();
+                result += Ui.taskNumberMessage(currList);
+                return result;
+            case "delete":
+                Task toBeDeleted = currList.getTaskAt(Integer.parseInt(parts[1]) - 1);
+                result += "Alrighty, this task's gone:\n";
+                result += toBeDeleted;
+                currList.removeTask(toBeDeleted);
+                return result;
+            case "find":
+                String searchWord = parts[1];
+                result += "Here are the matching tasks:\n";
+                TaskList matched = currList.matchingItems(searchWord);
+                for (int i = 0; i < matched.getLength(); i++) {
+                    result += i + 1 + ". " + matched.getTaskAt(i).toString() + "\n";
+                }
+                return result;
+            default:
+                return result;
+            }
+
```

Comment:
The block of code feels heavy, perhaps the logic could be spread out elsewhere, with only the key idea such as the case-switch statement being here 🤔 .

Reply:
Definitely possible to abstract out the logic, thanks for the suggestion.